App.js

```diff
@@ -26,68 +28,66 @@ export default class App extends React.Component {
       Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf"),
       ...Ionicons.font
     });
-    this.setState({ isReady: true });
-    this.setState({ screen: "map" });
+
+    this.setState({ isReady: true, screen: "map" });
   }
 
   load_user = () => {
-    if (this.state.searchedText.length > 0) {
+    const { searchedText } = this.state;
+
+    if (searchedText.length > 0) {
       this.setState({ isLoading: true });
-      catchUserTLFromId(this.state.searchedText).then(datas => {
+      getDataTLFromId(this.state.searchedText).then(datas => {
         this.setState({ user_tl: datas, isLoading: false });
       });
     }
   };
-  _searchTextInputChanged = text => {
-    this.setState({ searchedText: text });
+
+  searchTextInputChanged = text => {
+    this.setState({ searchFocus: true, searchedText: text });
   };
 
   change_screen_timeline = () => {
-    this.setState({ screen: "timeline" });
+    this.setState({ screen: "timeline", searchFocus: false });
   };
+
   change_screen_map = () => {
-    this.setState({ screen: "map" });
+    this.setState({ screen: "map", searchFocus: false });
   };
+
   change_screen_propos = () => {
-    this.setState({ screen: "propos" });
+    this.setState({ screen: "propos", searchFocus: false });
   };
+
   render() {
-    const { isReady, screen } = this.state;
+    const { isReady, screen, searchedText } = this.state;
     if (!isReady) {
       return <AppLoading />;
     }
-    let Component;
-    switch (screen) {
-      case "timeline":
-        Component = () => (
```

Comment:
This part was responsible for the bug. We do need the `() =>` part as we saw, but actually it causes re-render because it creates a new `Search` component every time.